workspaces/ui-v2/src/optic-components/pages/diffs/AddEndpointsPage/AddEndpointsPage.tsx

```diff
@@ -37,128 +26,131 @@ import { useChangelogStyles } from '../../../changelog/ChangelogBackground';
 import { useRunOnKeypress } from '<src>/optic-components/hooks/util';
 
 import {
+  AddEndpointDiffHeader,
+  BulkLearnModal,
   ExistingEndpointNameField,
   PendingEndpointNameField,
-} from './EndpointNameEditFields';
+  UndocumentedUrl,
+} from './components';
+import { useCheckboxState } from './hooks';
+
 import { useAnalytics } from '<src>/analytics';
 
-export function DiffUrlsPage(props: any) {
-  const urls = useUndocumentedUrls();
+export function DiffUrlsPage() {
+  const undocumentedUrls = useUndocumentedUrls();
   const history = useHistory();
-  const {
-    documentEndpoint,
-    stageEndpoint,
-    pendingEndpoints,
-  } = useSharedDiffContext();
+  const { documentEndpoint } = useSharedDiffContext();
   const diffReviewPagePendingEndpoint = useDiffReviewPagePendingEndpoint();
   const classes = useStyles();
   const analytics = useAnalytics();
+  const [showBulkModal, setShowBulkModal] = useState(false);
+  const [selectedUrls, setSelectedUrls] = useState<Set<string>>(new Set());
 
-  const [filteredUrls, setFilteredUrls] = useState(urls);
-
-  useEffect(() => {
-    setFilteredUrls(urls);
-  }, [pendingEndpoints.length, urls]);
-
+  const [searchQuery, setSearchQuery] = useState('');
   const [bulkMode, setBulkMode] = useState<boolean>(false);
-  const unmatchedUrlLengths = urls.filter((i) => !i.hide).length;
-
-  const shownUrls = filteredUrls.filter((i) => !i.hide);
-
-  function renderRow(props: ListChildComponentProps) {
-    const { index, style } = props;
-    const data = shownUrls[index];
-
-    return (
-      <UndocumentedUrl
-        style={style}
-        bulkMode={bulkMode}
-        {...data}
-        key={data.method + data.path}
-        onFinish={(pattern, method, autolearn) => {
-          const pendingId = documentEndpoint(pattern, method);
-          analytics.userDocumentedEndpoint(autolearn);
-          if (autolearn) {
-            stageEndpoint(pendingId);
-            setBulkMode(bulkMode);
-          } else {
-            const link = diffReviewPagePendingEndpoint.linkTo(pendingId);
-            history.push(link);
-          }
-        }}
-      />
-    );
-  }
+  const unmatchedUrlLengths = undocumentedUrls.length;
+  const visibleUrls = undocumentedUrls.filter((url) =>
+    url.path.startsWith(searchQuery)
+  );
+  const bulkSelectedUrls = undocumentedUrls.filter((url) =>
+    selectedUrls.has(url.path + url.method)
+  );
```

Comment:
This is slightly confusing - the reason we need this is we want to show:
- The total number of urls we have that are uncollapsed
- The visible urls (i.e. things with a filter applied)
- All the urls you've selected (excluding any urls that fall under a matched group, BUT including urls that have been filtered by the url filter)